common/shared.c

```diff
@@ -833,11 +833,14 @@ void ft_free_res(void)
 {
 	ft_close_fids();
 
-	free(tx_ctx_arr);
-	free(rx_ctx_arr);
-	tx_ctx_arr = NULL;
-	rx_ctx_arr = NULL;
-
+	if (tx_ctx_arr) {
+		free(tx_ctx_arr);
+		tx_ctx_arr = NULL;
+	}
+	if (rx_ctx_arr) {
+		free(rx_ctx_arr);
+		rx_ctx_arr = NULL;
+	}
```

Comment:
Yes, it's safe too. It was done just to align with "free" above. I'll revert this change